src/components/views/settings/tabs/user/SecurityUserSettingsTab.js

```diff
@@ -93,24 +97,62 @@ export default class SecurityUserSettingsTab extends React.Component {
         this.setState({ignoredUsers});
     };
 
-    _onRejectAllInvitesClicked = (rooms, ev) => {
+    _getInvitedRooms = () => {
+        return MatrixClientPeg.get().getRooms().filter((r) => {
+            return r.hasMembershipState(MatrixClientPeg.get().getUserId(), "invite");
+        });
+    };
+
+    _manageInvites = async (accept) => {
         this.setState({
-            rejectingInvites: true,
+            managingInvites: true,
         });
-        // reject the invites
-        const promises = rooms.map((room) => {
-            return MatrixClientPeg.get().leave(room.roomId).catch((e) => {
-                // purposefully drop errors to the floor: we'll just have a non-zero number on the UI
-                // after trying to reject all the invites.
-            });
+
+        // Compile array of invitation room ids
+        const invitedRoomIds = this._getInvitedRooms().map((room) => {
+            return room.roomId;
         });
-        Promise.all(promises).then(() => {
-            this.setState({
-                rejectingInvites: false,
+
+        // Execute all acceptances/rejections sequentially
+        const self = this;
+        const cli = MatrixClientPeg.get();
+        const action = accept ? cli.joinRoom.bind(cli) : cli.leave.bind(cli);
+        for (let i = 0; i < invitedRoomIds.length; i++) {
+            const roomId = invitedRoomIds[i];
+
+            // Accept/reject invite
+            await action(roomId).then(() => {
+                // No error, update invited rooms button
+                this.setState({invitedRoomAmt: self.state.invitedRoomAmt - 1});
+            }, async (e) => {
+                // Action failure
+                if (e.errcode === "M_LIMIT_EXCEEDED") {
+                    // Add a delay between each invite change in order to avoid rate
+                    // limiting by the server.
+                    await Promise.delay(e.retry_after_ms);
```

Comment:
This is an optional parameter - default to 2500ms?

Reply:
Ah really. Ok, good call.